CMakeLists.txt

```diff
@@ -127,48 +125,70 @@ set(HEADERS
         ${SYNC_BENCH_INC}/BenchParams.h
         ${SYNC_BENCH_INC}/DataObjectGenerator.h
         ${SYNC_BENCH_INC}/RandGen.h
-        ${SYNC_BENCH_INC}/FromFileGen.h)
+        ${SYNC_BENCH_INC}/FromFileGen.h
+    )
 
-include_directories(include)
-include_directories(tests)
-
-include_directories(/usr/local/include)
-link_directories(/usr/local/lib)
-link_directories(/usr/local/lib64)
-include_directories(${CMAKE_INSTALL_INCLUDEDIR})
-link_directories(${CMAKE_INSTALL_LIBDIR})
-
-# Add libs and executables
+# Add gensync library
 add_library(gensync STATIC ${SOURCE_FILES} ${HEADERS})
-target_include_directories(gensync PRIVATE ${CMAKE_SOURCE_DIR}/include)
-TARGET_LINK_LIBRARIES(gensync ntl cppunit pthread gmp)
+target_include_directories(gensync PUBLIC ${INCLUDE_DIR})
+target_compile_definitions(gensync PUBLIC RECORD="${RECORD_DIR}")
+target_compile_definitions(gensync PUBLIC DEFAULT_LOGLEVEL=${DEFAULT_LOG_LEVEL})
+
+# include Apache Data Sketches as a header-only library
+target_link_libraries(gensync PUBLIC ntl pthread gmp)
+target_include_directories(gensync PUBLIC
+                    incubator-datasketches-cpp/common/include
+                    incubator-datasketches-cpp/hll/include
+                    incubator-datasketches-cpp/fi/include
+    )
 
 install(CODE "FILE(REMOVE_RECURSE ${CMAKE_INSTALL_INCLUDEDIR}/gensync)") #Remove the existing GenSync library data and reinstall
 install(CODE "FILE(REMOVE_RECURSE ${CMAKE_INSTALL_LIBDIR}/libgensync.a)") #Remove the existing GenSync library data and reinstall
 install(TARGETS gensync LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install(DIRECTORY DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/GenSync) #create a new folder for GenSync inside of the system include directory
 install(DIRECTORY ${AUX_DIR_INC} ${DATA_DIR_INC} ${COMM_DIR_INC} ${SYNC_DIR_INC} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/GenSync COMPONENT devel) #copy the contents of the include folder into the system include directory
 
+# Add the TryMe executable
+add_executable(TryMe ${SRC_DIR}/TryMe.cpp)
+target_link_libraries(TryMe gensync)
+
+# Benchmarks
+add_executable(Benchmarks ${BENCH_DIR}/Runner.cpp)
+target_link_libraries(Benchmarks gensync)
+
+# Tools
+add_executable(EncodeJoin ${TOOLS_DIR}/EncodeJoin.cpp)
```

Comment:
What is EncodeJoin?

Reply:
I don't know what EncodeJoin is used for. It is added as executable in the gensync-benchmark CMakeLists.txt [here](https://github.com/nislab/gensync-benchmarking/blob/8e439aa9ae1e41b82c228d493a8ae45ca130f80b/CMakeLists.txt#L150), but this repository contains its source. I just added it for completeness.